### ISSUE TITLE
Allow more flexibility on locks/events

### ISSUE THREAD

**Issue by elast0ny** [elast0ny/mem_file#12]
For the  new API, it would be nice if users could create as many locks/events as they want for their shared memory.

For example, one might want to have 2 events for a single mapping which could allow you to do things like send/recv between two processes.

I'm thinking of something like : 
```Rust
//Prepare the mapping
my_mem = SharedMem::new("some_path", 4096).unwrap()
    .add_lock(LockType::Mutex, 0, 2047) // Lock to give you access to the first 2kb
    .add_lock(LockType::Mutex, 2048, -1) // Lock that gives access to the last 2kb
    .add_event();

//Create the mapping with all the requested features
my_mem.create().unwrap();

my_mem.wait(0); // wait on the first event
my_mem.rlock(1); // acquire lock on the last 2kb
```

**Comment by elast0ny** [comment]
Progress is being done in feature-mem-config branch

**Comment by elast0ny** [comment]
Now implemented in [feature-mem-config branch](https://github.com/elast0ny/shared_memory-rs/tree/feature-mem-config) for Windows/Linux.

Running into minor issure for OSX and maybe FreeBSD but will work on it in another issue.
